[exporter/prometheusremotewrite] Collector crashes while Scraping Histogram Metrics

**Describe the bug**
I'm running Opentelemetry collector contrib v0.56.0 Windows Amd64 binary with OTLP Receiver in Windows machine and I have a client application in Spring boot v2.6.9 using Micrometer Registry OTLP v1.9.2 to publish metrics to collector binary in OTLP Format. But as soon as client application starts publishing the metrics, the collector Crashes.
Related to #7149

**Steps to reproduce**
1. Start otel collector binary with provided config below.
2. From https://start.spring.io/ start a new Spring boot project with v2.6.9 and add below micrometer Dependency
```
    <dependency>
      <groupId>io.micrometer</groupId>
      <artifactId>micrometer-registry-otlp</artifactId>
      <version>1.9.2</version>
    </dependency>
```
3. Create a ```OTLPMeterRegistry``` Bean to start exposing the metrics in OTLP format and start the application.(The collector publish endpoint will be default i.e; http://localhost:4318/v1/metrics)
```
    @Bean
    public OtlpMeterRegistry otlpMeterRegistry(){
        return new OtlpMeterRegistry();
    }
```


**What did you expect to see?**
Metrics to be _remote written_ to the specified remote write endpoint.

**What did you see instead?**
Collector crashes with below exception
```
panic: runtime error: index out of range [-1]

goroutine 138 [running]:
go.opentelemetry.io/collector/pdata/internal.ImmutableUInt64Slice.At(...)
        go.opentelemetry.io/collector/pdata@v0.56.0/internal/generated_immutable_slice.go:125
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheusremotewrite.addSingleHistogramDataPoint({0xc0002df6d8?}, {0x4591ed?}, {0xc000b3acc0?}, {{0x0?, 0x40e2c5?}, 0xc000285aa0?}, 0x50?)
        github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheusremotewrite@v0.56.0/helper.go:346 +0xf65
github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheusremotewrite.FromMetrics({0x9?}, {{0x0?, 0xb3ebe1c?}, 0xc000285aa0?})
        github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheusremotewrite@v0.56.0/metrics_to_prw.go:88 +0x74a
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter.(*prwExporter).PushMetrics(0xc000147b90, {0x6179b98, 0xc000b41080}, {0x6179bd0?})
        github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusremotewriteexporter@v0.56.0/exporter.go:137 +0xd3
go.opentelemetry.io/collector/exporter/exporterhelper.(*metricsRequest).export(0x6179bd0?, {0x6179b98?, 0xc000b41080?})
        go.opentelemetry.io/collector@v0.56.0/exporter/exporterhelper/metrics.go:65 +0x34
go.opentelemetry.io/collector/exporter/exporterhelper.(*timeoutSender).send(0xc000462750, {0x618ecf0, 0xc0001624e0})
        go.opentelemetry.io/collector@v0.56.0/exporter/exporterhelper/common.go:225 +0x96
go.opentelemetry.io/collector/exporter/exporterhelper.(*retrySender).send(0xc000147c20, {0x618ecf0, 0xc0001624e0})
        go.opentelemetry.io/collector@v0.56.0/exporter/exporterhelper/queued_retry.go:176 +0x556
go.opentelemetry.io/collector/exporter/exporterhelper.(*metricsSenderWithObservability).send(0xc001018450, {0x618ecf0, 0xc0001624e0})
        go.opentelemetry.io/collector@v0.56.0/exporter/exporterhelper/metrics.go:132 +0x88
go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).start.func1({0x51e46e0?, 0xc0001624e0})
        go.opentelemetry.io/collector@v0.56.0/exporter/exporterhelper/queued_retry_inmemory.go:119 +0x5c
go.opentelemetry.io/collector/exporter/exporterhelper/internal.consumerFunc.consume(0xc0003a7200?, {0x51e46e0?, 0xc0001624e0?})
        go.opentelemetry.io/collector@v0.56.0/exporter/exporterhelper/internal/bounded_memory_queue.go:82 +0x2c
go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers.func2()
        go.opentelemetry.io/collector@v0.56.0/exporter/exporterhelper/internal/bounded_memory_queue.go:69 +0xcd
created by go.opentelemetry.io/collector/exporter/exporterhelper/internal.(*boundedMemoryQueue).StartConsumers
        go.opentelemetry.io/collector@v0.56.0/exporter/exporterhelper/internal/bounded_memory_queue.go:63 +0x98
```

**What version did you use?**
Version: 0.56.0

**What config did you use?**
Config: 
```
receivers:
  otlp:
    protocols:
      grpc:
      http:
  zipkin:

exporters:
  prometheusremotewrite:
    endpoint: "https://remotewriteurl"
  jaeger:
    endpoint: "some.jaeger.endpoint"
  prometheus:
    endpoint: "127.0.0.1:9091"
  file:
    path: otel-logs.txt

processors:
  batch:

extensions:
  health_check:
  pprof:
  zpages:

service:
  extensions: [ pprof, zpages, health_check ]
  pipelines:
    metrics:
      receivers: [ otlp ]
      exporters: [ prometheus, file, prometheusremotewrite ]
    traces:
      receivers: [ otlp,zipkin ]
      exporters: [ jaeger, file ]
  telemetry:
    logs:
      level: "debug"
```

**Environment**
OS: Windows

--- a/exporter/prometheusremotewriteexporter/exporter_test.go
+++ b/exporter/prometheusremotewriteexporter/exporter_test.go
@@ -369,6 +369,8 @@ func Test_PushMetrics(t *testing.T) {
 
 	histogramBatch := getMetricsFromMetricList(validMetrics1[validHistogram], validMetrics2[validHistogram])
 
+	emptyDataPointHistogramBatch := getMetricsFromMetricList(validMetrics1[validEmptyHistogram], validMetrics2[validEmptyHistogram])
+
 	summaryBatch := getMetricsFromMetricList(validMetrics1[validSummary], validMetrics2[validSummary])
 
 	// len(BucketCount) > len(ExplicitBounds)
@@ -468,6 +470,13 @@ func Test_PushMetrics(t *testing.T) {
 			metrics:            &histogramBatch,
 			reqTestFunc:        checkFunc,
 			expectedTimeSeries: 12,
+			httpResponseCode:   http.StatusAccepted,
+		},
+		{
+			name:               "valid_empty_histogram_case",
+			metrics:            &emptyDataPointHistogramBatch,
+			reqTestFunc:        checkFunc,
+			expectedTimeSeries: 6,
 			httpResponseCode:   http.StatusAccepted,
 		},
 		{

--- a/exporter/prometheusremotewriteexporter/testutil_test.go
+++ b/exporter/prometheusremotewriteexporter/testutil_test.go
@@ -56,13 +56,14 @@ var (
 	quantileValues = []float64{7, 8, 9}
 	quantiles      = getQuantiles(quantileBounds, quantileValues)
 
-	validIntGauge    = "valid_IntGauge"
-	validDoubleGauge = "valid_DoubleGauge"
-	validIntSum      = "valid_IntSum"
-	validSum         = "valid_Sum"
-	validHistogram   = "valid_Histogram"
-	validSummary     = "valid_Summary"
-	suffixedCounter  = "valid_IntSum_total"
+	validIntGauge       = "valid_IntGauge"
+	validDoubleGauge    = "valid_DoubleGauge"
+	validIntSum         = "valid_IntSum"
+	validSum            = "valid_Sum"
+	validHistogram      = "valid_Histogram"
+	validEmptyHistogram = "valid_empty_Histogram"
+	validSummary        = "valid_Summary"
+	suffixedCounter     = "valid_IntSum_total"
 
 	validIntGaugeDirty = "*valid_IntGauge$"
 
@@ -70,22 +71,24 @@ var (
 
 	// valid metrics as input should not return error
 	validMetrics1 = map[string]pmetric.Metric{
-		validIntGauge:    getIntGaugeMetric(validIntGauge, lbs1, intVal1, time1),
-		validDoubleGauge: getDoubleGaugeMetric(validDoubleGauge, lbs1, floatVal1, time1),
-		validIntSum:      getIntSumMetric(validIntSum, lbs1, intVal1, time1),
-		suffixedCounter:  getIntSumMetric(suffixedCounter, lbs1, intVal1, time1),
-		validSum:         getSumMetric(validSum, lbs1, floatVal1, time1),
-		validHistogram:   getHistogramMetric(validHistogram, lbs1, time1, floatVal1, uint64(intVal1), bounds, buckets),
-		validSummary:     getSummaryMetric(validSummary, lbs1, time1, floatVal1, uint64(intVal1), quantiles),
+		validIntGauge:       getIntGaugeMetric(validIntGauge, lbs1, intVal1, time1),
+		validDoubleGauge:    getDoubleGaugeMetric(validDoubleGauge, lbs1, floatVal1, time1),
+		validIntSum:         getIntSumMetric(validIntSum, lbs1, intVal1, time1),
+		suffixedCounter:     getIntSumMetric(suffixedCounter, lbs1, intVal1, time1),
+		validSum:            getSumMetric(validSum, lbs1, floatVal1, time1),
+		validHistogram:      getHistogramMetric(validHistogram, lbs1, time1, floatVal1, uint64(intVal1), bounds, buckets),
+		validEmptyHistogram: getHistogramMetricEmptyDataPoint(validEmptyHistogram, lbs1, time1),
+		validSummary:        getSummaryMetric(validSummary, lbs1, time1, floatVal1, uint64(intVal1), quantiles),
 	}
 	validMetrics2 = map[string]pmetric.Metric{
-		validIntGauge:      getIntGaugeMetric(validIntGauge, lbs2, intVal2, time2),
-		validDoubleGauge:   getDoubleGaugeMetric(validDoubleGauge, lbs2, floatVal2, time2),
-		validIntSum:        getIntSumMetric(validIntSum, lbs2, intVal2, time2),
-		validSum:           getSumMetric(validSum, lbs2, floatVal2, time2),
-		validHistogram:     getHistogramMetric(validHistogram, lbs2, time2, floatVal2, uint64(intVal2), bounds, buckets),
-		validSummary:       getSummaryMetric(validSummary, lbs2, time2, floatVal2, uint64(intVal2), quantiles),
-		validIntGaugeDirty: getIntGaugeMetric(validIntGaugeDirty, lbs1, intVal1, time1),
+		validIntGauge:       getIntGaugeMetric(validIntGauge, lbs2, intVal2, time2),
+		validDoubleGauge:    getDoubleGaugeMetric(validDoubleGauge, lbs2, floatVal2, time2),
+		validIntSum:         getIntSumMetric(validIntSum, lbs2, intVal2, time2),
+		validSum:            getSumMetric(validSum, lbs2, floatVal2, time2),
+		validHistogram:      getHistogramMetric(validHistogram, lbs2, time2, floatVal2, uint64(intVal2), bounds, buckets),
+		validEmptyHistogram: getHistogramMetricEmptyDataPoint(validEmptyHistogram, lbs2, time2),
+		validSummary:        getSummaryMetric(validSummary, lbs2, time2, floatVal2, uint64(intVal2), quantiles),
+		validIntGaugeDirty:  getIntGaugeMetric(validIntGaugeDirty, lbs1, intVal1, time1),
 		unmatchedBoundBucketHist: getHistogramMetric(unmatchedBoundBucketHist, pcommon.NewMap(), 0, 0, 0,
 			pcommon.NewImmutableFloat64Slice([]float64{0.1, 0.2, 0.3}),
 			pcommon.NewImmutableUInt64Slice([]uint64{1, 2})),
@@ -290,6 +293,17 @@ func getEmptyCumulativeHistogramMetric(name string) pmetric.Metric {
 	metric.SetName(name)
 	metric.SetDataType(pmetric.MetricDataTypeHistogram)
 	metric.Histogram().SetAggregationTemporality(pmetric.MetricAggregationTemporalityCumulative)
+	return metric
+}
+
+func getHistogramMetricEmptyDataPoint(name string, attributes pcommon.Map, ts uint64) pmetric.Metric {
+	metric := pmetric.NewMetric()
+	metric.SetName(name)
+	metric.SetDataType(pmetric.MetricDataTypeHistogram)
+	metric.Histogram().SetAggregationTemporality(pmetric.MetricAggregationTemporalityCumulative)
+	dp := metric.Histogram().DataPoints().AppendEmpty()
+	attributes.CopyTo(dp.Attributes())
+	dp.SetTimestamp(pcommon.Timestamp(ts))
 	return metric
 }
 

--- a/pkg/translator/prometheusremotewrite/helper.go
+++ b/pkg/translator/prometheusremotewrite/helper.go
@@ -343,7 +343,9 @@ func addSingleHistogramDataPoint(pt pmetric.HistogramDataPoint, resource pcommon
 	if pt.Flags().HasFlag(pmetric.MetricDataPointFlagNoRecordedValue) {
 		infBucket.Value = math.Float64frombits(value.StaleNaN)
 	} else {
-		cumulativeCount += pt.BucketCounts().At(pt.BucketCounts().Len() - 1)
+		if pt.BucketCounts().Len() > 0 {
+			cumulativeCount += pt.BucketCounts().At(pt.BucketCounts().Len() - 1)
+		}
 		infBucket.Value = float64(cumulativeCount)
 	}
 	infLabels := createAttributes(resource, pt.Attributes(), settings.ExternalLabels, nameStr, baseName+bucketStr, leStr, pInfStr)

--- a/unreleased/prw-no-hist-panic.yaml
+++ b/unreleased/prw-no-hist-panic.yaml
@@ -1,0 +1,11 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: exporter/prometheusremotewrite
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Fix a panic when a histogram does not have any buckets"
+
+# One or more tracking issues related to the change
+issues: [12777]

